server/src/main/java/org/apache/iotdb/db/qp/strategy/optimizer/RemoveNotOptimizer.java

```diff
@@ -41,7 +43,10 @@
    */
   @Override
   public FilterOperator optimize(FilterOperator filter) throws LogicalOperatorException {
-    return removeNot(filter);
+    Set<PartialPath> pathSet = filter.getPathSet();
+    FilterOperator optimizedFilterOperator = removeNot(filter);
+    optimizedFilterOperator.setPathSet(pathSet);
+    return optimizedFilterOperator;
```

Comment:
It seems `pathSet` is not changed in removeNot() method, why reseting it?

Reply:
Previously this code actually appeared in `optimizeSFWOperator`... Removing the code would cause tests failed... The reason is that the `removeNot` may not return the original `OptimizedFilterOperator`.